test/Quake/dependency-if-bug-lowering.qke

```diff
@@ -0,0 +1,49 @@
+// ========================================================================== //
+// Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                 //
+// All rights reserved.                                                       //
+//                                                                            //
+// This source code and the accompanying materials are made available under   //
+// the terms of the Apache License 2.0 which accompanies this distribution.   //
+// ========================================================================== //
+
+// RUN: cudaq-opt --add-wireset --assign-wire-indices --dep-analysis %s | FileCheck %s
+
+func.func @__nvqpp__mlirgen__run_test() -> i1 attributes {"cudaq-entrypoint", "cudaq-kernel"} {
+  %true = arith.constant true
+  %0 = quake.null_wire
+  %1 = quake.null_wire
+  %2:2 = cc.if(%true) ((%arg0 = %0, %arg1 = %1)) -> (!quake.wire, !quake.wire) {
+    %4 = quake.h %arg0 : (!quake.wire) -> !quake.wire
+    %5 = quake.x %4 : (!quake.wire) -> !quake.wire
+    %6 = quake.h %arg1 : (!quake.wire) -> !quake.wire
+    cc.continue %5, %6 : !quake.wire, !quake.wire
+  } else {
+    cc.continue %arg0, %arg1 : !quake.wire, !quake.wire
+  }
+  %measOut, %wires = quake.mz %2#0 name "b" : (!quake.wire) -> (!quake.measure, !quake.wire)
+  %3 = quake.discriminate %measOut : (!quake.measure) -> i1
+  quake.sink %wires : !quake.wire
+  quake.sink %2#1 : !quake.wire
+  return %3 : i1
+}
+
+// CHECK-LABEL:   quake.wire_set @wires[2147483647] attributes {sym_visibility = "private"}
+
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__run_test() -> i1 attributes {"cudaq-entrypoint", "cudaq-kernel"} {
+// CHECK:           %[[VAL_0:.*]] = arith.constant true
+// CHECK:           %[[VAL_1:.*]] = quake.borrow_wire @wires[1] : !quake.wire
+// CHECK:           %[[VAL_2:.*]] = quake.borrow_wire @wires[0] : !quake.wire
```

Comment:
Should these 2 lines be `CHECK-DAG`, assuming the order is irrelevant?

Reply:
A lot of the tests could have `CHECK-DAG`, but even `CHECK-DAG` is a bit restrictive with the reordering that is possible with `dep-analysis`, so it's likely that any change to the ordering will break the tests in other places too.